src/main/scala/viper/gobra/frontend/info/implementation/property/TypeMerging.scala

```diff
@@ -18,7 +18,17 @@ trait TypeMerging extends BaseProperty { this: TypeInfoImpl =>
     // currently, only merging of l and r being identical types is supported
     // possible future improvement: if l is assignable to r, then return r instead of None (or vice versa)
     (l, r) match {
-      case (Single(lst), Single(rst)) => if (identicalTypes(lst, rst)) Some(lst) else None
+      case (Single(lst), Single(rst)) =>
+        if (identicalTypes(lst, rst)) Some(lst) else {
+          (lst, rst) match {
+            case (a, IntT(UntypedConst)) if underlyingType(a).isInstanceOf[IntT] => Some(a)
+            case (IntT(UntypedConst), b) if underlyingType(b).isInstanceOf[IntT] => Some(b)
+            case (SequenceT(l), SequenceT(r)) => for {
```

Comment:
What is with the other data structure kinds, e.g. sets, multisets, array, structs, ...

Reply:
I'm not sure if it makes sense to define this function for structs given that they are statically defined and any struct literal must have the fields of the declared type. I think that it makes sense for all other types that you mentioned.

Reply:
Check 84706b9